tutorials/cloud-functions-firestore/index.md

```diff
@@ -0,0 +1,256 @@
+
+    npm init
+
+### Install @google-cloud/firestore
+
+Let's add the firestore client to our node app, saving the dependency.
+
+    npm install --save --save-exact @google-cloud/firestore
+
+### Writing the Function Code
+
+You can copy and paste the simplified version of the code,
+into `index.js` (or
+[download](https://raw.githubusercontent.com/GoogleCloudPlatform/community/master/tutorials/cloud-functions-firestore/index.js)).
+
+    const Firestore = require('@google-cloud/firestore');
+    const PROJECTID = 'cloud-functions-firestore';
+    const COLLECTION_NAME = 'cloud-functions-firestore';
+    const firestore = new Firestore({
+      projectId: PROJECTID,
+      timestampsInSnapshots: true,
+    });
+
+    exports.main = (req, res) => {
+      if (req.method === 'DELETE') throw 'not yet built';
+      if (req.method === 'POST') {
+        // store/insert a new document
+        const data = (req.body) || {};
+        const ttl = Number.parseInt(data.ttl);
+        const ciphertext = (data.ciphertext || '').replace(/[^a-zA-Z0-9\-]*/g, '');
+        const created = new Date().getTime();
+        return firestore.collection(COLLECTION_NAME)
+          .add({ created, ttl, ciphertext })
+          .then(doc => {
+            return res.status(200).send(doc);
+          }).catch(err => {
+            console.error(err);
+            return res.status(404).send({ error: 'unable to store', err });
+          });
+      }
+      // read/retrieve an existing document by id
+      if (!(req.query && req.query.id)) {
+        return res.status(404).send({ error: 'No Id' });
+      }
+      const id = req.query.id.replace(/[^a-zA-Z0-9]/g, '').trim();
+      if (!(id && id.length)) {
+        return res.status(404).send({ error: 'Empty Id' });
+      }
+      return firestore.collection(COLLECTION_NAME)
+        .doc(id)
+        .get()
+        .then(doc => {
+          if (!(doc && doc.exists)) {
+            return res.status(404).send({ error: 'Unable to find the document' });
+          }
+          const data = doc.data();
+          return res.status(200).send(data);
+        }).catch(err => {
+          console.error(err);
+          return res.status(404).send({ error: 'Unable to retrieve the document' });
+        });
+    };
+
+In that code, we are going to listen for a POST
+with the data fields of `ciphertext` and `ttl`
+and store the values into a new document in firestore
+using the `add()` function (which auto-assigns an id).
+
+And we are also listening for a GET
+and if we have an `id` in the querystring,
+we will lookup that document in firestore and
+return the document if found.
+
+NOTE: you probably want more input sanitation for a production application.
+
+## Deploying the Function
+
+You now have a `package.json` file listing your dependencies
+and you have an `index.js` file which will listen for a HTTP trigger.
+
+We want to make google automatically run that code for us, when needed.
```

Comment:
Capitalize 'Google'. Also, which part of Google, exactly?

Reply:
using "Google Cloud Functions"